runtime/src/accounts.rs

```diff
@@ -550,49 +557,97 @@ impl Accounts {
-    pub fn commit_credits(&self, ancestors: &HashMap<Fork, usize>, fork: Fork) {
+    pub fn commit_credits_and_rents(
+        &self,
+        rent_collector: &RentCollector,
+        ancestors: &HashMap<Fork, usize>,
+        fork: Fork,
+    ) -> u64 {
         // Clear the credit only hashmap so that no further transactions can modify it
         let credit_only_account_locks = Self::take_credit_only(&self.credit_only_account_locks)
-            .expect("Credit only locks didn't exist in commit_credits");
-        self.store_credit_only_credits(credit_only_account_locks, ancestors, fork);
+            .expect("Credit only locks didn't exist in commit_credits_and_rents");
+        self.store_credit_only_credits_and_rents(
+            credit_only_account_locks,
+            rent_collector,
+            ancestors,
+            fork,
+        )
     }
 
     /// Used only for tests to store credit-only accounts after every transaction
-    pub fn commit_credits_unsafe(&self, ancestors: &HashMap<Fork, usize>, fork: Fork) {
+    pub fn commit_credits_and_rents_unsafe(
+        &self,
+        rent_collector: &RentCollector,
+        ancestors: &HashMap<Fork, usize>,
+        fork: Fork,
+    ) -> u64 {
         // Clear the credit only hashmap so that no further transactions can modify it
         let mut w_credit_only_account_locks = self.credit_only_account_locks.write().unwrap();
         let w_credit_only_account_locks =
             Self::get_write_access_credit_only(&mut w_credit_only_account_locks)
                 .expect("Credit only locks didn't exist in commit_credits");
-        self.store_credit_only_credits(w_credit_only_account_locks.drain(), ancestors, fork);
+        self.store_credit_only_credits_and_rents(
+            w_credit_only_account_locks.drain(),
+            rent_collector,
+            ancestors,
+            fork,
+        )
     }
 
-    fn store_credit_only_credits<I>(
+    fn store_credit_only_credits_and_rents<I>(
         &self,
         credit_only_account_locks: I,
+        rent_collector: &RentCollector,
         ancestors: &HashMap<Fork, usize>,
         fork: Fork,
-    ) where
+    ) -> u64
+    where
         I: IntoIterator<Item = (Pubkey, CreditOnlyLock)>,
     {
-        for (pubkey, lock) in credit_only_account_locks {
-            let lock_count = *lock.lock_count.lock().unwrap();
-            if lock_count != 0 {
-                warn!(
-                    "dropping credit-only lock on {}, still has {} locks",
-                    pubkey, lock_count
-                );
-            }
-            let credit = lock.credits.load(Ordering::Relaxed);
-            if credit > 0 {
-                let mut account = self
-                    .load_slow(ancestors, &pubkey)
-                    .map(|(account, _)| account)
-                    .unwrap_or_default();
+        let mut accounts: HashMap<Pubkey, Account> = HashMap::new();
```

Comment:
why a hashmap?

Reply:
Initial reasoning was to avoid duplication here. But, I guess you are right. No need for hashmap here.

Reply:
done